### ISSUE TITLE
Hey, some questions.

### ISSUE THREAD

**Issue by WildDylan** [devxoul/URLNavigator#33]
VC name `A` want push VC name `B`. If `A` want get some informations from `B`, `B` will contain a block property that `A` set it, `B` call it. How to do this with `URLNavigator` ?

**Comment by devxoul** [comment]
@WildDylan, Hi!

Use `URLNavigator.viewController(for:)` to get view controller instance from URL and set a property. Then push it via `URLNavigator`.

**A.swift**

```swift
if let b: B = Navigator.viewController(for: "myapp://b") {
  b.propertyBlock = {
    return ...
  }
  Navigator.push(b, from: self) // `self` is VC named A
}
```

**Comment by WildDylan** [comment]
@devxoul As you say, i need B.Type. But now, we only know `myapp://b` this scheme.

**Comment by WildDylan** [comment]
Before commit #36 ,`Navigator ` only can send [String: String] to destination ViewController. Now, You can add `Any` type param when added `parameters` param to push and present method. Like this:
```swift
Navigator.push(URL, ["backedMessage": { print($0) }])
```
and use this call back:
```swift
let callBack = values["backedMessage"] as! YourBackedFunctionType
callBack("Hello world!")
```
I've add this code to demo of `Navigator `. Thanks for your reading.

**Comment by devxoul** [comment]
@WildDylan, could you please provide me some example situation? I cannot understand why do we need this because without reference of `B`, you cannot map this view controller to URLNavigator. It will be great If you can share an example project.

**Comment by WildDylan** [comment]
Haha,  This is not my original idea that you understand.
A lot of times, property in `ViewController` should be private. #36 means, with `Any type` params to initialize the ViewController before push or present.

And your quesiton: `I cannot understand why do we need this because without reference of B, you cannot map this view controller to URLNavigator`. About this, because many many projects are developed by a team. Members have their own `Module`. `Router` as the project middleware managed mapping、actions between `ViewControllers`.

Why we need Router? Because When `Module A` want's open `Module B `'s ViewController, only know ViewController's `URL` in `Module B` is enough. But your question said that `How to map it without reference of B`, `Module B` no need to care about what `Module A` mapped. `Module A`'s VC was mapped in `Module A`; `Module B`'s VC was mapped in `Module B`. If we want open, just in `Module B` call `Navigator.push("moduleA://detail")`.

Before this comment, i have said: `map with className`, but i think it's not safe. 
#36 only means, add extra param `[string: Any]` is useful. Thanks for your response.
And example project in #36 have tried this.

**Comment by devxoul** [comment]
@WildDylan, Thank you for your explanation. That makes sense. Let's move to #36 to talk about implementation :)

**Comment by WildDylan** [comment]
Closed with #36
